packages/lib-core/src/store/PluginLoader.ts

```diff
@@ -304,10 +343,146 @@ export class PluginLoader {
     });
   }
 
+  /**
+   * Append entry module to DOM
+   */
+  injectPluginScript(
+    baseURL: string,
+    pluginLoadData: PluginLoadData,
+    // onLoad type does not have stand alone definition
+    onLoad?: (ev?: Event) => void,
+    onError?: OnErrorEventHandler,
+    getDocument: () => typeof document = _.constant(document),
+  ) {
+    const pluginName = pluginLoadData.manifest.name;
+
+    this.plugins.set(pluginName, pluginLoadData);
+    const initPromise = new Promise<void>((resolve, reject) => {
+      const script = getDocument().createElement('script');
+
+      script.src = resolveURL(baseURL, pluginLoadData.manifest.entryScript);
+      script.async = true;
+
+      script.onload = async (event) => {
+        if (onLoad) {
+          onLoad(event);
+        }
+        if (pluginLoadData.manifest.libraryType === 'var') {
+          await this.initializeVarModule(pluginName);
+        }
+        return resolve();
+      };
+
+      script.onerror = (event) => {
+        if (onError) {
+          onError(event);
+        }
+        return reject(new Error('Unable to initialize remote module'));
+      };
+
+      getDocument().head.appendChild(script);
+
+      consoleLogger.info(`Loading entry script of plugin ${pluginName} from ${script.src}`);
+    });
+
+    return initPromise;
+  }
+
+  /**
+   * Initialize container loaded to window scope.
+   *
+   * Required by HCC plugins.
+   * Simulates the jsonp flow used in the other type of plugins.
+   */
+  async initializeVarModule(pluginName: string) {
+    if (!this.plugins.has(pluginName)) {
+      throw new ErrorWithCause(`No plugin data exist for '${pluginName}'`);
+    }
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const data = this.plugins.get(pluginName)!;
+    const moduleScope = data.manifest.name;
```

Comment:
Plugin name is required and will always be present in a valid plugin manifest.

In other words, `pluginName` should be equal to `data.manifest.name`.

Thus, we should be able to remove the "else" part below.